gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenAuthorityService.java

```diff
@@ -184,7 +191,7 @@ public boolean verifyToken(JWT token, RSAPublicKey publicKey)
     PublicKey key;
     try {
       if (publicKey == null) {
-        key = ks.getSigningKeystore().getCertificate(getSigningKeyAlias(signingKeyAlias)).getPublicKey();
+        key = ks.getSigningKeystore().getCertificate(getSigningKeyAlias(null)).getPublicKey();
```

Comment:
Is `null` correct here? That seems weird...

Reply:
In the original code, `signingKeyAlias` was set to `config.getSigningKeyAlias()`. 

In the new code `getSigningKeyAlias(String signingKeystoreAlias)` is set to return `signingKeystoreAlias` if `signingKeystoreAlias` is not null or `config.getSigningKeyAlias()` if `signingKeystoreAlias` is null.  

This is the same thing.  I don't particularly like the code, but I was trying not to change too much.  Since nothing is broken, I left it.

Reply:
Ok I'll pull the code down tomorrow into an IDE and take a look to see the flow.

Reply:
How about something like this:
```
private String getSigningKeyAlias(String signingKeystoreAlias) {
    if(signingKeystoreAlias != null) {
     return signingKeystoreAlias;
    }

    // Fallback to defaults
    return getSigningKeyAlias();
  }

  private String getSigningKeyAlias() {
    String alias = config.getSigningKeyAlias();
    return (alias == null) ? GatewayConfig.DEFAULT_SIGNING_KEY_ALIAS : alias;
  }
```

The places where `getSigningKeyAlias(null)` can be changed to `getSigningKeyAlias()`

Reply:
sure.. I will update